releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java

```diff
@@ -224,236 +391,315 @@ public void testGenerateCustomTemplate() throws Exception {
-        final List<String> errors = MainProcess.validateNotes(
-            createNotes(null, null, null, createAllNotes(), null), createBaseCliOptions("1.0.1")
-                .setValidateVersion(true).build());
+        TEST_ISSUES.addAll(createAllIssues(MISCELLANEOUS_LABEL));
+        TEST_COMMITS.addAll(createAllCommits());
 
-        Assert.assertEquals("no errors", 0, errors.size());
+        runMainAndAssertReturnCode(0,
+            "-localRepoPath", getOutputLocation(),
+            "-remoteRepoPath", "checkstyle/checkstyle",
+            "-startRef", "12345678",
+            "-releaseNumber", "1.0.1",
+            "-outputLocation", getOutputLocation(),
+            "-githubAuthToken", "TOKEN",
+            "-generateGitHub",
+            "-validateVersion"
+        );
     }
 
     @Test
     public void testValidateNotesPatchNewModule() throws Exception {
-        final List<String> errors = MainProcess.validateNotes(
-            createNotes(null, null, null, null, createAllNotes()), createBaseCliOptions("1.0.1")
-                .setValidateVersion(true).build());
+        TEST_ISSUES.addAll(createAllIssues(NEW_MODULE_LABEL));
+        TEST_COMMITS.addAll(createAllCommits());
 
-        Assert.assertEquals("one error", 1, errors.size());
-        Assert.assertEquals(errors.get(0), MSG_RELEASE_IS_PATCH);
+        runMainAndAssertReturnCode(-2,
+            "-localRepoPath", getOutputLocation(),
+            "-remoteRepoPath", "checkstyle/checkstyle",
+            "-startRef", "12345678",
+            "-releaseNumber", "1.0.1",
+            "-outputLocation", getOutputLocation(),
+            "-githubAuthToken", "TOKEN",
+            "-generateGitHub",
+            "-validateVersion"
+        );
     }
 
     @Test
     public void testValidateNotesPatchNewFeatureNewModuleAndBreaking() throws Exception {
-        final List<String> errors = MainProcess.validateNotes(
-            createNotes(createAllNotes(), createAllNotes(),
-                null, null, createAllNotes()), createBaseCliOptions("1.0.1")
-                .setValidateVersion(true).build());
-
-        Assert.assertEquals("one error", 1, errors.size());
-        Assert.assertEquals(errors.get(0), MSG_RELEASE_IS_PATCH);
+        TEST_ISSUES.addAll(createAllIssues(NEW_MODULE_LABEL));
+        TEST_ISSUES.addAll(createAllIssues(NEW_MODULE_LABEL));
+        TEST_ISSUES.addAll(createAllIssues(BREAKING_COMPATIBILITY_LABEL));
+        TEST_COMMITS.addAll(createAllCommits());
+
+        runMainAndAssertReturnCode(-2,
+            "-localRepoPath", getOutputLocation(),
+            "-remoteRepoPath", "checkstyle/checkstyle",
+            "-startRef", "12345678",
+            "-releaseNumber", "1.0.1",
+            "-outputLocation", getOutputLocation(),
+            "-githubAuthToken", "TOKEN",
+            "-generateGitHub",
+            "-validateVersion"
+        );
     }
 
     @Test
     public void testValidateNotesPatchBugAndMiscellaneous() throws Exception {
```

Comment:
Not going to lie, not a big fan of those `...ValidateNotes...` tests being here in `TemplateProcessorTest`. Nothing to do with testing the `TemplateProcessor`. I would personally put them in their own class like `ValidateNotesTest` or at least in `MainTest`. Let me know what you think.

Reply:
TemplateProcessorTest should "validate/test" only output file creation.

we have validation in MainProcess.validateNotes method.
so test file should be MainProcessTest

but execution of test happens from Main class, but it is detail of implementation, some execution might be directly by MainProcess.validateNotes.